app/models/hackbot/interactions/set_club.rb

```diff
@@ -0,0 +1,128 @@
+module Hackbot
+  module Interactions
+    class SetClub < Command
+      include Concerns::LeaderAssociable
+
+      TRIGGER = /set-club ?(?<stage_name>.+)/
+
+      USAGE = 'set-club <dormant|active>'.freeze
+      DESCRIPTION = 'pause check-ins for your club during summer break'.freeze
+
+      STAGES = {
+        'active' => '5003',
+        'dormant' => '5014'
+      }.freeze
+
+      def should_handle?
+        super && event[:user] == data['slack_id'] unless state == 'start'
+      end
+
+      # rubocop:disable Metrics/MethodLength
+      # rubocop:disable Metrics/AbcSize
+      def start
+        data['stage_name'] = captured[:stage_name]
+        data['stage_key'] = STAGES[data['stage_name']]
+        data['slack_id'] = event[:user]
+
+        invalid_stage && return unless data['stage_key']
+
+        case leader.clubs.count
+        when 0
+          msg_channel copy('no_clubs_found')
+        when 1
+          data['club'] = leader.clubs.first
+          set_single_club
+        else
+          set_one_in_many_clubs
+        end
+      end
+      # rubocop:enable Metrics/MethodLength
+      # rubocop:enable Metrics/AbcSize
+
+      def wait_for_club_select
+        return :wait_for_club_select unless action
+
+        data['club'] = Club.find action[:value]
+
+        send_action_result copy('many_clubs.action_result',
+                                club_name: data['club'].name)
+
+        set_single_club
+      end
+
+      # rubocop:disable Metrics/AbcSize
+      def wait_for_activation_date
+        return :wait_for_club_select unless msg
+
+        data['activation_date'] = Chronic.parse(msg, bias: :future)
+        human_date = data['activation_date']
+                     .to_date
+                     .to_formatted_s :long_ordinal
+        msg_channel copy('activation_date.confirmation',
+                         activation_date: human_date)
+
+        data['club'].update(activation_date: data['activation_date'])
+
+        :finish
+      end
+      # rubocop:enable Metrics/AbcSize
+
+      private
+
+      def set_one_in_many_clubs
+        actions = leader.clubs.map { |c| { text: c.name, value: c.id } }
+
+        msg_channel(
+          text: copy('many_clubs.prompt', stage_name: data['stage_name']),
+          attachments: [
+            actions: actions
+          ]
+        )
+
+        :wait_for_club_select
+      end
+
+      # rubocop:disable Metrics/AbcSize
+      def set_single_club
```

Comment:
Want to bring up my concerns here again:

This is a different method of setting the stage key than we used in the `CheckIn` class. In the `CheckIn` class we used the `make_dormant` and `make_active` helper functions which I wrote.